test/functional/pdo_sqlsrv/pdo_passwords.phpt

```diff
@@ -1,67 +1,66 @@
 --TEST--
 Test password with non alphanumeric characters
 --DESCRIPTION--
-The first three cases should have no problem connecting. Only the last case fails because the 
+The first three cases should have no problem connecting. Only the last case fails because the
 right curly brace should be escaped with another right brace.
 In Azure for this test to pass do not specify any particular database when connecting
 --SKIPIF--
-<?php require('skipif.inc'); ?>
+<?php require('skipif_mid-refactor.inc'); ?>
 --FILE--
 <?php
-  
-require  'MsSetup.inc';
+require_once("MsSetup.inc");
 
-try{
+$dsn = "sqlsrv:Server=$server;ConnectionPooling=false;";
+// get DSN information for Column Encryption
+if ($keystore != "none") {
+    $dsn .= "ColumnEncryption=Enabled;";
+}
+if ($keystore == "ksp") {
+    require('AE_Ksp.inc');
+    $ksp_path = getKSPPath();
+    $dsn .= "CEKeystoreProvider=$ksp_path;CEKeystoreName=$ksp_name;CEKeystoreEncryptKey=$encrypt_key;";
+}
+
+try {
     // Test 1
-    $conn = new PDO( "sqlsrv:Server=$server;ConnectionPooling=false;" , "test_password", "! ;4triou"); 
-    if(!$conn)
-    {
+    $conn = new PDO($dsn, "test_password", "! ;4triou");
+    if (!$conn) {
```

Comment:
this test reveals that we might consider adding another connect() method that takes not the default values from `MsSetup`
ideally it's better to maintain / contain the AE specific code in one place

Reply:
I think instead of having another connect method, I'll have one that returns a dsn string.